### PR TITLE
Combine RemoteNode and ProtocolHandler

### PR DESCRIPTION
**Pull request by erikzhang**

### CODE REVIEW

File: src/neo/Network/P2P/Connection.cs [hunk 1]
```diff
@@ -89,8 +89,8 @@ protected override void OnReceive(object message)
         {
             switch (message)
             {
-                case Timer _:
-                    Disconnect(true);
+                case Close close:
+                    Disconnect(close.Abort);
```
Comment (vncoelho) [inline src/neo/Network/P2P/Connection.cs]: When do we current use `Disconnect(false)` or just `Disconnect()`;

**Comment by Tommo-L** [comment]
I think it's not a good idea to merge them, it's too big to clearly read. 😂

**Comment by erikzhang** [comment]
I can split it into two files, but it must be in the same class. @Tommo-L

**Comment by Tommo-L** [comment]
Okay, it's good for me.

**Comment by erikzhang** [comment]
Done.

File: src/neo/Network/P2P/RemoteNode.ProtocolHandler.cs [hunk 2]
```diff
@@ -158,20 +131,17 @@ private void OnAddrMessageReceived(AddrPayload payload)
...
         private void OnFilterClearMessageReceived()
         {
             bloom_filter = null;
-            Context.Parent.Tell(new SetFilter { Filter = null });
         }
 
         private void OnFilterLoadMessageReceived(FilterLoadPayload payload)
         {
             bloom_filter = new BloomFilter(payload.Filter.Length * 8, payload.K, payload.Tweak, payload.Filter);
-            Context.Parent.Tell(new SetFilter { Filter = bloom_filter });
```
Comment (vncoelho) [inline src/neo/Network/P2P/RemoteNode.ProtocolHandler.cs:174]: Good simplification.
@Tommo-L, perhaps two files was a good move. Nice suggestion.

**Comment by shargon** [comment]
I like this PR :) simpler it's better, we are testing it

**Comment by erikzhang** [comment]
@shargon Are you still testing?

**Pull request merged**